CONTRIBUTING.md

```diff
@@ -1,81 +1,126 @@
+or you can contact Shinji Watanabe <shinjiw@ieee.org> or other main developers. 
+We shall discuss the design and implementation.
+Once we agree that the plan looks good, go ahead and implement it.
+You can find ongoing major development plans at https://github.com/espnet/espnet/milestones
+
+### 1.2 Minor Updates (minor feature, bug-fix for an issue)
+
+If you want to implement a minor feature or bug-fix for an issue, please first take a look at 
+the existing [issues](https://github.com/espnet/espnet/pulls) and/or [pull requests](https://github.com/espnet/espnet/pulls).
+Pick an issue and comment on the task that you want to work on this feature.
+If you need more context on a particular issue, please ask us, and then we shall provide more information.
    
-   We also welcome if you find some bugs during your actual use of ESPnet and make a PR to fix them.
+We also welcome if you find some bugs during your actual use of ESPnet and make a PR to fix them.
+
+### 1.3 Recipes
 
-3. ESPnet provides and maintains a lot of reproducible examples similar to Kaldi (called `recipe`).
-   The recipe creation/update/bug-fix is one of our major development items, and we really encourage 
-   you to work on it.
-   When you port a Kaldi recipe to ESPnet, see https://github.com/espnet/espnet/wiki/How-to-port-the-Kaldi-recipe-to-the-ESPnet-recipe%3F 
+ESPnet provides and maintains a lot of reproducible examples similar to Kaldi (called `recipe`).
+The recipe creation/update/bug-fix is one of our major development items, and we really encourage 
+you to work on it.
+
+ESPnet currently supports two versions of recipes: `egs` and `egs2` for espnet and espnet2, respectively.
+Each of them follows a different structure.
+
+#### 1.3.1 ESPnet1 recipes
+
+ESPnet1's recipes correspond to `egs`.ESPnet1 follows convention from [Kaldi](https://github.com/kaldi-asr/kaldi), and it is also based on several
+utilities from Kaldi. This feature naturally makes it very simple to port a Kaldi recipe to ESPnet. If you port a Kaldi recipe to ESPnet, please see
+[port-kaldi-recipe](https://github.com/espnet/espnet/wiki/How-to-port-the-Kaldi-recipe-to-the-ESPnet-recipe%3F) for more detailed instructions.
+If there is no existing Kaldi's recipe, you could still refer to 
+[port-kaldi-recipe](https://github.com/espnet/espnet/wiki/How-to-port-the-Kaldi-recipe-to-the-ESPnet-recipe%3F) and the major task is to prepare
+a Kaldi-style directory (Please see [Prepare-Kaldi-Style-Directory](https://kaldi-asr.org/doc/data_prep.html) for details.)
    
-   We also encourage you to report your results with your detailed environmental info and upload the model for the reproducibility 
-   (e.g., see https://github.com/espnet/espnet/blob/master/egs/tedlium2/asr1/RESULTS.md).
+For each recipe, we also encourage you to report your results with your detailed environmental info and upload the model for the reproducibility 
```

Comment:
For each recipe, we ask you to report the following: minimal results, detailed environmental and model info and upload the model for the reproducibility when it's possible. All these information should be written in a markdown file called `RESULTS.md` and put at the recipe root. You can refer to [tedlium2-example](https://github.com/espnet/espnet/blob/master/egs/tedlium2/asr1/RESULTS.md) for an example.